src/methods.rs

```diff
@@ -30,16 +33,37 @@ declare_lint!(pub WRONG_SELF_CONVENTION, Warn,
 declare_lint!(pub WRONG_PUB_SELF_CONVENTION, Allow,
               "defining a public method named with an established prefix (like \"into_\") that takes \
                `self` with the wrong convention");
+declare_lint!(pub CALLING_MAIN, Warn,
+              "calling into main detected");
 
 impl LintPass for MethodsPass {
     fn get_lints(&self) -> LintArray {
         lint_array!(OPTION_UNWRAP_USED, RESULT_UNWRAP_USED, STR_TO_STRING, STRING_TO_STRING,
-                    SHOULD_IMPLEMENT_TRAIT, WRONG_SELF_CONVENTION)
+                    SHOULD_IMPLEMENT_TRAIT, WRONG_SELF_CONVENTION, CALLING_MAIN)
     }
 }
 
 impl LateLintPass for MethodsPass {
     fn check_expr(&mut self, cx: &LateContext, expr: &Expr) {
+        fn no_std_present(cx: &LateContext) -> bool {
+        	let attrs : &[Attribute] = &cx.krate.attrs; // TODO replace these two...
+        	for attr in attrs { // lines with ...
+        	// for &attr in cx.krate.attrs { ... once type inf on nightly fixed
+                if let MetaWord(ref mword) = attr.node.value.node {
+                    if mword == &"no_std" { return true; }
+                }
+            }
+            false
+        }
+        if let ExprCall(ref func, ref args) = expr.node {
+            if args.is_empty() {
+                if let ExprPath(_, path) = func.node.clone() {
+                    if !path.global && path.segments[0].identifier.name.as_str() == "main" && !no_std_present(cx) {
```

Comment:
I could swear `.borrow().get(..)` is exactly exactly what I've tried and it didn't work, but, yes, it does, thanks. (I will continue to be very slow with this in the foreseeable future. I'd like to finish this ticket for the learning experience but if you guys just want it done I won't be upset if someone else picks it up.)